frontend/projects/upgrade/src/app/core/feature-flags/store/feature-flags.effects.ts

```diff
@@ -177,19 +179,81 @@ export class FeatureFlagsEffects {
 
   fetchFeatureFlagById$ = createEffect(() =>
     this.actions$.pipe(
-      ofType(FeatureFlagsActions.actionFetchFeatureFlagById),
+      ofType(featureFlagsActions.actionFetchFeatureFlagById),
       map((action) => action.featureFlagId),
       filter((featureFlagId) => !!featureFlagId),
       switchMap((featureFlagId) =>
         this.featureFlagsDataService.fetchFeatureFlagById(featureFlagId).pipe(
           map((data: FeatureFlag) => {
-            return FeatureFlagsActions.actionFetchFeatureFlagByIdSuccess({ flag: data });
+            return featureFlagsActions.actionFetchFeatureFlagByIdSuccess({ flag: data });
           }),
-          catchError(() => [FeatureFlagsActions.actionFetchFeatureFlagByIdFailure()])
+          catchError(() => [featureFlagsActions.actionFetchFeatureFlagByIdFailure()])
         )
       )
     )
   );
 
+  emailFeatureFlagData$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(featureFlagsActions.actionEmailFeatureFlagData),
+      map((action) => ({ featureFlagId: action.featureFlagId })),
+      withLatestFrom(this.store$.pipe(select(selectCurrentUser))),
+      filter(([{ featureFlagId }, { email }]) => !!featureFlagId && !!email),
+      switchMap(([{ featureFlagId }, { email }]) =>
+        this.featureFlagsDataService.emailFeatureFlagData(featureFlagId, email).pipe(
+          tap(() => {
+            email
+              ? this.notificationService.showSuccess(`Email will be sent to ${email}`)
+              : this.notificationService.showSuccess('Email will be sent to registered email');
+          }),
+          map(() => featureFlagsActions.actionEmailFeatureFlagDataSuccess()),
+          catchError(() => [featureFlagsActions.actionEmailFeatureFlagDataFailure()])
+        )
+      )
+    )
+  );
+
+  exportFeatureFlagsDesign$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(featureFlagsActions.actionExportFeatureFlagDesign),
+      map((action) => ({ featureFlagIds: action.featureFlagIds })),
+      filter(({ featureFlagIds }) => !!featureFlagIds),
+      switchMap(({ featureFlagIds }) =>
+        this.featureFlagsDataService.exportFeatureFlagsDesign(featureFlagIds).pipe(
+          tap(() => {
```

Comment:
the `tap` seems unnecesary for same reason as above, that call can just go in the `map` with the rest of the success handling.